geo-data-server/src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -236,6 +236,48 @@ void saveLines(List<LineGeoData> linesGeoData) {
         lineRepository.saveAll(linesEntities);
     }
 
+    private LineGeoData getLineGeoDataWithEndSubstations(Map<String, LineGeoData> linesGeoDataDb, Map<String, SubstationGeoData> substationGeoDataDb, Line line) {
+        LineGeoData geoData = linesGeoDataDb.get(line.getId());
+        Substation sub1 = line.getTerminal1().getVoltageLevel().getSubstation();
+        Substation sub2 = line.getTerminal2().getVoltageLevel().getSubstation();
+        Coordinate substation1Coordinate = substationGeoDataDb.get(sub1.getId()).getCoordinate();
+        Coordinate substation2Coordinate = substationGeoDataDb.get(sub2.getId()).getCoordinate();
+        if (geoData == null || geoData.getCoordinates().isEmpty()) {
+            return new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(),
+                List.of(substation1Coordinate, substation2Coordinate));
+        }
+        // now we are sure that there is at last 1 coordinate in geodata coordinate
+        if (distance(geoData.getCoordinates().get(0), substation1Coordinate) >
+            distance(geoData.getCoordinates().get(0), substation2Coordinate)) {
+            LineGeoData fullLine = new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(),
+                addCoordinates(substation2Coordinate, geoData.getCoordinates(), substation1Coordinate));
+            Collections.reverse(fullLine.getCoordinates());
+            return fullLine;
+        } else {
+            return new LineGeoData(line.getId(), sub1.getNullableCountry(), sub2.getNullableCountry(),
+                addCoordinates(substation1Coordinate, geoData.getCoordinates(), substation2Coordinate));
+        }
+
+    }
+
+    private static double distance(Coordinate p1, Coordinate p2) {
```

Comment:
We shouldn't use the distance between substations and the first pylon to determine which substations goes to each side. Instead, the geo-data-server should use the available network from the network-store to know which substation should go where